modules/library/main/src/main/java/org/geotools/filter/ExpressionDOMParser.java

```diff
@@ -63,17 +64,17 @@
     /** number of coordinates in a box */
     private static final int NUM_BOX_COORDS = 5;
 
-    /** Creates a new instance of ExpressionXmlParser */
-    private ExpressionDOMParser() {
-        this(CommonFactoryFinder.getFilterFactory2(null));
-        LOGGER.finer("made new logic factory");
-    }
     /** Constructor injection */
     public ExpressionDOMParser(FilterFactory2 factory) {
         ff = factory != null ? factory : CommonFactoryFinder.getFilterFactory2(null);
     }
-    /** Setter injection */
+
+    /** Setter injection
+     * @deprecated Will not be API in the future anymore.
+     */
+    @Deprecated
```

Comment:
I don't think anything is using it, but there are other setters for filter factory, one quite prominent and widely used in ContentDataStore.  Several setters of the same kind are used in GeoServer as well.

In order to deprecate an API we need some arguments about why it's bad (and why it's bad here, but maybe ok in other parts of the code?). Maybe part of the reasoning is in your message to the devel list, need to check it better (wans'nt clear to me at a first read).

Reply:
Not sure if I've understood correctly. Should I move discussion to developer mailing list?
IMHO its a bit strange to have both constructor argements and setters for injection. In this specific case every usage of filterfactory would need an additional null check to keep safe and an plan what to do in case it is null. I assume it should throw a ParserException or something simmilar to indicate that parsing without a filterFactory doesn't make any sense, espeacially because ExpressionDomParser has a fallback anyway.

Having this in mind it might be another option to remove FilterFactory2 usage in SLDParser and initialite ExpressionDomParser with null. or even make this formar private constructor public. I do not really like the first aproach because developers need to have knowlege about internal implementation.

Suggestions?

Reply:
The null check you added is a good thing, should be enough (I see the constructor has its own way of dealing with it).
The rest seems up to personal preference, I would not deprecate a method based just on it, but I don't feel strongly about it either. If another core developer is happy to vouch for deprecation, I'm not going to oppose.